apps/site/pages/docs/api-extensions.mdx

```diff
@@ -1,463 +1,127 @@
 
-It's important to know that FastStore exposes the data that comes from FastStore API along with FastStore API Extensions inside a provider called `PageProvider`.
-This data comes as a `context` in `PageProvider`, and you can use some hooks to access it:
+### Fetch only the data you need
+GraphQL allows you to customize your query to get precisely the data you need, which reduces the data returned from the requests. Double-check your queries to make sure they are not over-fetching.
 
-The `usePDP()` hook is recommendable in cases where you use your section on PDP.
+This also means that you must be mindful of the data brought into your query by [GraphQL fragments](https://graphql.org/learn/queries/#fragments). Use them only when they do not include unnecessary data.
 
-```ts copy
-import { usePDP } from "@faststore/core"
 
-const context = usePDP()
-```
+### Do not send requests to other APIs
 
-The `usePLP()` hook is recommendable in cases where you use your section on PLP.
+When building your storefront with FastStore, to avoid comprimising the website performance and lead to `504` timeout errors, do not send requests to APIs other than the FastStore API.
+If you need to access other data not available in the native FastStore API schema, you must do this by using [API extension](/docs/api-extensions). *See the code's comment*
 
-```ts copy
-import { usePLP } from "@faststore/core"
+{/*WILL E CONTINUE WITH THIS RECOMMENDATION? If you need data that is not available on the native [FastStore GraphQL schema](https://v1.faststore.dev/reference/api/objects), check the [root object](https://v1.faststore.dev/reference/api/objects).
+The [GraphQL root object](https://graphql.org/learn/execution/#root-fields-resolvers) represents the top-level object returned to a query. You can see what data is available in the root object by adding logs.*/}
 
-const context = usePLP()
-```
 
-The `useSearchPage()` hook is recommendable in cases where you use your section on the Search Page.
+### Avoid custom API Routes
+We do not recomend using Next.js [API Routes](https://nextjs.org/docs/api-routes/introduction) other than the ones already built in FastStore.
 
-```ts copy
-import { useSearchPage } from "@faststore/core"
+If you do use this type of function, you must be mindful to always check which ones are actually running and remove the ones that are not.
 
-const context = useSearchPage()
-```
+To see what API Routes are running in your project, you can... *(See the code's comment)*  {/* IS THE FOLLOWING STILL VALID? run the command `yarn build` at the root of your project and check the log. Then, check the directory `src/pages/api` of your project and remove functions that are not running or are not included in the [starter](/starters) you used. */}
```

Comment:
It is not valid. In fact, API routes are no longer available to developers, so they can't add new ones. I think we can remove this recommendation or make this limitation (that we chose to have) clear in another section. WDYT?